src/lib/operators/table_scan/column_vs_value_table_scan_impl.cpp

```diff
@@ -127,6 +131,121 @@ void ColumnVsValueTableScanImpl::_scan_segment(const BaseDictionarySegment& segm
   });
 }
 
+void ColumnVsValueTableScanImpl::_scan_sorted_segment(const BaseSegment& segment, const ChunkID chunk_id,
+                                                      PosList& matches,
+                                                      const std::shared_ptr<const PosList>& position_filter) const {
+  resolve_data_and_segment_type(segment, [&](const auto type, const auto& typed_segment) {
+    if constexpr (std::is_same_v<decltype(typed_segment), const ReferenceSegment&>) {
+      Fail("Expected ReferenceSegments to be handled before calling this method");
+    } else {
+      // using ColumnDataType = typename decltype(type)::type;
+      // auto typed_value = type_cast_variant<ColumnDataType>(_value);
+
+      Assert(segment.sort_order().value() == OrderByMode::AscendingNullsLast ||
+                 segment.sort_order().value() == OrderByMode::Ascending ||
+                 segment.sort_order().value() == OrderByMode::DescendingNullsLast ||
+                 segment.sort_order().value() == OrderByMode::Descending,
+             "Unsupported sort type");
+
+      // TODO(hendraet): Support Null values correctly
+      auto segment_iterable = create_iterable_from_segment(typed_segment);
+      segment_iterable.with_iterators(position_filter, [&](auto begin, auto end) {
+        auto bounds = get_sorted_bounds(position_filter, begin, end, typed_segment);
+        auto lower_it = std::get<0>(bounds);
+        auto upper_it = std::get<1>(bounds);
+        auto exclude_range = std::get<2>(bounds);
+
+        if (exclude_range) {
+          for (; begin != lower_it; ++begin) {
+            const auto& value = *begin;
+            matches.emplace_back(RowID{chunk_id, value.chunk_offset()});
+          }
+          std::advance(begin, std::distance(lower_it, upper_it));
+          for (; begin != end; ++begin) {
+            const auto& value = *begin;
+            matches.emplace_back(RowID{chunk_id, value.chunk_offset()});
+          }
+        } else {
+          for (; lower_it != upper_it; ++lower_it) {
+            const auto& value = *lower_it;
+            matches.emplace_back(RowID{chunk_id, value.chunk_offset()});
```

Comment:
Is there a chance to use something like memcopy here when we move large consecutively stores position lists?

Reply:
While we couldn't think of a way to use `memcopy` here, using `resize()` instead of `reserve()` has lead to a speedup of about 5x: https://github.com/hyrise/hyrise/pull/1394/commits/0f338c820646dacbb925a38c2f4469a1f5f91498